compiler/src/test/scala/org/scalajs/core/compiler/test/JSInteropTest.scala

```diff
@@ -750,4 +750,381 @@ class JSInteropTest extends DirectTest with TestHelpers {
 
   }
 
+  @Test
+  def scalaJSDefinedJSNameOverrideWarnings: Unit = {
+    """
+    @ScalaJSDefined
+    abstract class A extends js.Object {
+      def bar(): Int
+    }
+    @ScalaJSDefined
+    class B extends A {
+      override def bar() = 1
+    }
+    """.hasNoWarns
+
+    """
+    @ScalaJSDefined
+    abstract class A extends js.Object {
+      @JSName("foo")
+      def bar(): Int
+    }
+    @ScalaJSDefined
+    class B extends A {
+      @JSName("foo")
+      override def bar() = 1
+    }
+    """.hasNoWarns
+
+    """
+    @ScalaJSDefined
+    trait A extends js.Object {
+      @JSName("foo")
+      def bar(): Int
+    }
+    @ScalaJSDefined
+    class B extends A {
+      @JSName("foo")
+      override def bar() = 1
+    }
+    """.hasNoWarns
+
+    """
+    @ScalaJSDefined
+    abstract class A extends js.Object {
+      @JSName("foo")
+      def bar(): Int
+    }
+    @ScalaJSDefined
+    class B extends A {
+      @JSName("baz")
+      override def bar() = 1
+    }
+    """ hasWarns
+    """
+      |newSource1.scala:11: warning:
+      |A Scala.js-defined JS class member is overriding another with a different exported name.
+      |
+      |override def bar(): Int at line 13 with exported JSName 'baz'
+      |    is conflicting with
+      |def bar(): Int in class A with exported JSName 'foo'
+      |
+      |    class B extends A {
+      |          ^
```

Comment:
For example of the position problem, ideally this should point to `B.bar`, not to `B` itself.